homeassistant/components/uptime_kuma/sensor.py

```diff
@@ -0,0 +1,182 @@
+)
+from homeassistant.const import CONF_URL, EntityCategory, UnitOfTime
+from homeassistant.core import HomeAssistant, callback
+from homeassistant.helpers.device_registry import DeviceEntryType, DeviceInfo
+from homeassistant.helpers.entity_platform import AddConfigEntryEntitiesCallback
+from homeassistant.helpers.typing import StateType
+from homeassistant.helpers.update_coordinator import CoordinatorEntity
+
+from .const import DOMAIN, HAS_CERT, HAS_HOST, HAS_PORT, HAS_URL
+from .coordinator import UptimeKumaConfigEntry, UptimeKumaDataUpdateCoordinator
+
+PARALLEL_UPDATES = 0
+
+
+class UptimeKumaSensor(StrEnum):
+    """Uptime Kuma sensors."""
+
+    CERT_DAYS_REMAINING = "cert_days_remaining"
+    RESPONSE_TIME = "response_time"
+    STATUS = "status"
+    TYPE = "type"
+    URL = "url"
+    HOSTNAME = "hostname"
+    PORT = "port"
+
+
+@dataclass(kw_only=True, frozen=True)
+class UptimeKumaSensorEntityDescription(SensorEntityDescription):
+    """Uptime Kuma sensor description."""
+
+    value_fn: Callable[[UptimeKumaMonitor], StateType]
+    create_entity: Callable[[MonitorType], bool]
+
+
+SENSOR_DESCRIPTIONS: tuple[UptimeKumaSensorEntityDescription, ...] = (
+    UptimeKumaSensorEntityDescription(
+        key=UptimeKumaSensor.CERT_DAYS_REMAINING,
+        translation_key=UptimeKumaSensor.CERT_DAYS_REMAINING,
+        device_class=SensorDeviceClass.DURATION,
+        native_unit_of_measurement=UnitOfTime.DAYS,
+        value_fn=lambda m: m.monitor_cert_days_remaining,
+        create_entity=lambda t: t in HAS_CERT,
+    ),
+    UptimeKumaSensorEntityDescription(
+        key=UptimeKumaSensor.RESPONSE_TIME,
+        translation_key=UptimeKumaSensor.RESPONSE_TIME,
+        device_class=SensorDeviceClass.DURATION,
+        native_unit_of_measurement=UnitOfTime.MILLISECONDS,
+        value_fn=(
+            lambda m: m.monitor_response_time if m.monitor_response_time > -1 else None
+        ),
+        create_entity=lambda _: True,
+    ),
+    UptimeKumaSensorEntityDescription(
+        key=UptimeKumaSensor.STATUS,
+        translation_key=UptimeKumaSensor.STATUS,
+        device_class=SensorDeviceClass.ENUM,
+        options=[m.name.lower() for m in MonitorStatus],
+        value_fn=lambda m: m.monitor_status.name.lower(),
+        create_entity=lambda _: True,
+    ),
+    UptimeKumaSensorEntityDescription(
+        key=UptimeKumaSensor.TYPE,
+        translation_key=UptimeKumaSensor.TYPE,
+        device_class=SensorDeviceClass.ENUM,
+        options=[m.name.lower() for m in MonitorType],
+        value_fn=lambda m: m.monitor_type.name.lower(),
+        entity_category=EntityCategory.DIAGNOSTIC,
+        create_entity=lambda _: True,
+    ),
+    UptimeKumaSensorEntityDescription(
+        key=UptimeKumaSensor.URL,
+        translation_key=UptimeKumaSensor.URL,
+        entity_category=EntityCategory.DIAGNOSTIC,
+        value_fn=lambda m: m.monitor_url,
+        create_entity=lambda t: t in HAS_URL,
+    ),
```

Comment:
I am wondering if this has any added bonus, as this is already in the device registry, so people can already automate with that